app/jobs/cache_analytics_job.rb

```diff
@@ -5,38 +5,27 @@ class CacheAnalyticsJob < ApplicationJob
   queue_as :default
 
   def perform(*_args)
-    Analytic.transaction do
-      analytics_by_jur_path = {}
-      root_jurs = []
+    Jurisdiction.all.find_each do |jur|
+      Analytic.transaction do
+        analytic = Analytic.new(jurisdiction_id: jur.id)
 
-      leaf_nodes = Jurisdiction.all.select { |jur| jur.has_children? == false }
-      leaf_nodes.each do |leaf_jurisdiction|
-        leaf_analytic = self.class.calculate_analytic_local_to_jurisdiction(leaf_jurisdiction)
-        analytics_by_jur_path[leaf_jurisdiction[:path]] = leaf_analytic
-        # Start recursive bubble up of analytic data
-        self.class.add_analytic_to_parent(leaf_jurisdiction, leaf_analytic, analytics_by_jur_path)
-      end
-
-      # Map data will be on the top-level jurisdiction only
-      root_nodes = Jurisdiction.where(ancestry: nil)
-      root_nodes.each do |root_jurisdiction|
-        symp_by_state = root_jurisdiction.all_patients.pluck(:address_state).each_with_object(Hash.new(0)) { |state, counts| counts[state] += 1 }
-        monitored_by_state = root_jurisdiction.all_patients.symptomatic.uniq.pluck(:address_state).each_with_object(Hash.new(0)) do |state, counts|
-          counts[state] += 1
+        # Map data will be on the top-level jurisdiction only (will be removed once new maps are implemented)
+        if jur.root?
```

Comment:
Would this prevent, for example, "State 1" from being able to use the maps?

Reply:
So these are the old/current maps that are being used, which only get populated in the root jurisdictions like so:

https://github.com/SaraAlert/SaraAlert/blob/master/app/jobs/cache_analytics_job.rb#L22

When other jurisdiction analysts visit the analytics page, these maps are retrieved from the analytic of the root jurisdiction like so:

https://github.com/SaraAlert/SaraAlert/blob/master/app/controllers/analytics_controller.rb#L59

So to answer your question, this does not prevent State 1 analysts from seeing these maps.

I've purposely kept these in here for this PR because David's new maps are not ready yet but I will add a commit to his branch at some point to remove these old maps

Reply:
I see, that makes perfect sense. I think I was confused because I was thinking about _where_ we are going to go with this versus how it is _currently_. Thanks!